src/r_draw.c

```diff
@@ -823,6 +821,12 @@ void R_InitBuffer(int width, int height)
 
   // Column offset. For windows.
 
+  // [Nugget] Dynamic arrays
+  if (first_allocation)
```

Comment:
Shouldn't be necessary, just make sure arrays point to `NULL` when uninitialized. `realloc(NULL, ...)` is supposed to be the same as `malloc(...)`.